manifests/module.pp

```diff
@@ -9,75 +9,114 @@
 # 
 # @example compile and load the apache module
 #   selinux::module{ 'apache':
-#     ensure => 'present',
-#     source => 'puppet:///modules/selinux/apache.te',
+#     ensure    => 'present',
+#     source_te => 'puppet:///modules/selinux/apache.te',
+#     builder   => 'simple'
 #   }
 #
 # @param ensure present or absent
-# @param sx_mod_dir path where source is stored and the module built. 
-#   Valid values: absolute path
-# @param source the source file (either a puppet URI or local file) of the SELinux .te file
-# @param content content of the source .te file
-# @param makefile absolute path to the selinux-devel Makefile
-# @param prefix (DEPRECATED) the prefix to add to the loaded module. Defaults to ''.
-#   Does not work with CentOS >= 7.2 and Fedora >= 24 SELinux tools.
+# @param source_te the source file (either a puppet URI or local file) of the SELinux .te file
+# @param source_fc the source file (either a puppet URI or local file) of the SELinux .fc file
+# @param source_if the source file (either a puppet URI or local file) of the SELinux .if file
+# @param builder either 'simple' or 'refpolicy'. The simple builder attempts to use checkmodule
+# to build the module, whereas 'refpolicy' uses the refpolicy framework, but requires 'make'
 # @param syncversion selmodule syncversion param
 define selinux::module(
-  $source       = undef,
-  $content      = undef,
-  $ensure       = 'present',
-  $makefile     = '/usr/share/selinux/devel/Makefile',
-  $prefix       = '',
-  $sx_mod_dir   = '/usr/share/selinux',
+  Optional[String] $source_te = undef,
+  Optional[String] $source_fc = undef,
+  Optional[String] $source_if = undef,
+  Enum['absent', 'present'] $ensure = 'present',
+  Optional[Enum['simple', 'refpolicy']] $builder = undef,
   $syncversion  = undef,
 ) {
 
+  if $builder == 'refpolicy' {
+    require ::selinux::refpolicy_package
```

Comment:
can we maybe make this `selinux::package::policy_devel`? the term refpolicy related to  the package isn't quite right because refpolicy is what Tresys provides. The RHEL/Fedora fork has large differences to the refpolicy. 

But maybe at all not important and just a personal opinion of mine. :)